src/bin/stratis-min/stratisd-min.rs

```diff
@@ -27,13 +41,82 @@ fn parse_args() -> App<'static, 'static> {
         )
 }
 
-fn main() -> Result<(), Box<dyn Error>> {
-    let mut app = parse_args();
+fn get_long_help(app: &mut App) -> Result<String, Box<dyn Error>> {
     let mut help = Vec::new();
     app.write_long_help(&mut help)?;
-    let help = String::from_utf8(help)?;
+    Ok(String::from_utf8(help)?)
+}
+
+/// To ensure only one instance of stratisd runs at a time, acquire an
+/// exclusive lock. Return an error if lock attempt fails.
+fn trylock_pid_file() -> StratisResult<File> {
+    let mut f = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(STRATISD_MIN_PID_PATH)
+        .map_err(|err| {
+            StratisError::Error(format!(
+                "Failed to create or open the stratisd-min PID file at {}: {}",
+                STRATISD_MIN_PID_PATH, err
+            ))
+        })?;
+    let stratisd_min_file = match flock(f.as_raw_fd(), FlockArg::LockExclusiveNonblock) {
+        Ok(_) => {
+            f.write_all(getpid().to_string().as_bytes())?;
+            Ok(f)
+        }
+        Err(_) => {
+            let mut buf = String::new();
+
+            if f.read_to_string(&mut buf).is_err() {
+                buf = "<unreadable>".to_string();
+            }
+
+            Err(StratisError::Error(format!(
+                "Daemon already running with supposed pid: {}",
+                buf
+            )))
+        }
+    };
```

Comment:
Should it return immediately if it can't get the lock on the "stratisd-min" file instead of continuing and attempting the lock on the stratisd file?

Reply:
Yes, that actually makes more sense. I'll push an edit.